### ISSUE TITLE
`odo push` with alternate --run-command should push complete file set upon new pod creation

### ISSUE THREAD

**Issue by scottkurz** [openshift/odo#3918]
/kind bug

## What versions of software are you using?

**Operating System:** Win10

**Output of `odo version`:**   This is close to 1.2.6, with https://github.com/openshift/odo/pull/3478 built in

## How did you run odo exactly?

Roughly the workflow is:

1. $ odo create java-openliberty --starter
2. $ odo push
3.  <edit a .java file>
4. $ odo push --run-command run-test-off

I say "roughly" because with some devfile syntax tweaks the starter devfile keeps getting broken at a fixed level of 'odo' by devfile changes.   So you might have to adjust this a bit to recreate.

## Actual behavior

With the new run command, the deployment spec changes, so k8s starts a new pod, but the problem is that only the one .java file I changed gets copied into the new container, not the whole project.   My new container only has:  ` /projects/src/...my.java` not the pom.xml even to build it.   I can force the whole proj. to get copied with `odo push -f` but that's not obvious.

## Expected behavior

Since this is a new pod, 'odo' should sync the entire project into the new pod, not just the delta/changed file set.

## Any logs, error output, etc?

```
Executing devfile commands for component java-openliberty
 <  Executing devbuild command "if [ -e /projects/.disable-bld-cmd ]; then echo \"found the disable file\" && echo \"devBuild command will not run\" && exit 0; else echo \"will run the devBuild command\" && echo \"...moving liberty\" && mkdir -p /projects/target/liberty  
 
 ...

  X  Executing devbuild command "if [ -e /projects/.disable-bld-cmd ]; then echo \"found the disable file\" && echo \"devBuild command will not run\" && exit 0; else echo \"will run the devBuild command\" && echo \"...moving liberty\" && mkdir -p /projects/target/liberty && mvn -Dmaven.repo.local=/mvn/repository package && mv /opt/ol/wlp /projects/target/liberty && touch ./.disable-bld-cmd; fi", if not running [4s]
 X  Failed to start component with name java-openliberty. Error: Failed to create the component: command execution failed: unable to execute the run command: unable to exec command [/bin/sh -c cd /projects && if [ -e /projects/.disable-bld-cmd ]; then echo "found the disable file" && echo "devBuild command will not run" && exit 0; else echo "will run the devBuild command" && echo "...moving liberty" && mkdir -p /projects/target/liberty && mvn -Dmaven.repo.local=/mvn/repository package && mv /opt/ol/wlp /projects/target/liberty && touch ./.disable-bld-cmd; fi]:
will run the devBuild command
...moving liberty
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.111 s
[INFO] Finished at: 2020-09-09T11:26:16Z
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/projects). Please verify you invoked Maven from the correct directory. -> [Help 1]
```

I could zip up the `-v 9` logs if it helps.

**Comment by johnmcollier** [comment]
I can't reproduce @scottkurz 😕 

Can you check to see if the branch of odo you're using has the fix from #3838 in it? This sounds similar to the bug that it fixed. Basically, we weren't detecting that the component pod had restarted and that the source code needed to be re-synced. It looks like a similar thing is happening here. 

Because we're changing the run command, the run command env var in the container is changing and the container gets restarted, but we're not properly detecting that the container has restarted, and not force-syncing as required. But I would expect the fix from #3838 to catch this.

**Comment by scottkurz** [comment]
> Can you check to see if the branch of odo you're using has the fix from #3838 in it (commit id [a7408ed](https://github.com/openshift/odo/commit/a7408ed9b0535ddcf1f123bf856391b5cc7424bb))?

Guessing not based on the dates.  I'll wait and try this on the beta release.  Thanks.

**Comment by scottkurz** [comment]
Seems fixed on the odo v2.0.0-beta-1.
